### PR TITLE
Changes for OpenAIRE Funders API v2

### LINKED ISSUE #2816: Search/Import Funded Projects from OpenAIRE API to create new Entities
From release plan; assigned to Dimitris at OpenAIRE. 
Estimate: 60 hours
DEPENDS ON "Convert External Source to Entity" FEATURE.  Would require a new external source to support searching the OpenAIRE API

### PR DESCRIPTION
**Pull request by dimitrispie**
## Description
OpenAIRE Funders API as External Provider

Fixes #2816

### CODE REVIEW

**Comment by paulo-graca** [comment]
@dimitrispie you have some build failures. Might be related with style checking. Please check:
https://github.com/DSpace/DSpace/runs/1574931469#step:5:843

File: dspace-api/pom.xml [hunk 1]
```diff
@@ -899,7 +899,10 @@
             <artifactId>bcel</artifactId>
             <version>6.4.0</version>
         </dependency>
-
-    </dependencies>
+<dependency>
```
Comment (paulo-graca) [inline dspace-api/pom.xml:902]: There is an indentation issue.

**Comment by dimitrispie** [comment]
Done

**Comment by paulo-graca** [comment]
Hi Dimitris, Am I missing something? I'm getting this error when trying to access OpenAIRE Projects external source:
`{"timestamp":"2020-12-18T14:03:39.468+0000","status":400,"error":"Bad Request","message":"Request is invalid or incorrect","path":"/server/api/integration/externalsources/openaireAPI/entries"}
`

I also have this in my external-services.xml file, it should work:
```
    <bean class="org.dspace.external.provider.impl.OpenAIREFunderDataProvider" init-method="init">
         <constructor-arg value="${openaire.api.url}"/>
         <property name="sourceIdentifier" value="openaireAPI"/>
     </bean>   
```

and my Server side has a:

```
{
  "id": "openaireAPI",
  "name": "openaireAPI",
  "hierarchical": false,
  "type": "externalsource",
  "_links": {
    "self": {
      "href": "https://myhost/server/api/integration/externalsources/openaireAPI"
    },
    "entries": {
      "href": "https://myhost/server/api/integration/externalsources/openaireAPI/entries?query="
    }
  }
}
```

**Comment by dimitrispie** [comment]
> Hi Dimitris, Am I missing something? I'm getting this error when trying to access OpenAIRE Projects external source:
> `{"timestamp":"2020-12-18T14:03:39.468+0000","status":400,"error":"Bad Request","message":"Request is invalid or incorrect","path":"/server/api/integration/externalsources/openaireAPI/entries"} `
> 
> I also have this in my external-services.xml file, it should work:
> 
> ```
>     <bean class="org.dspace.external.provider.impl.OpenAIREFunderDataProvider" init-method="init">
>          <constructor-arg value="${openaire.api.url}"/>
>          <property name="sourceIdentifier" value="openaireAPI"/>
>      </bean>   
> ```
> 
> and my Server side has a:
> 
> ```
> {
>   "id": "openaireAPI",
>   "name": "openaireAPI",
>   "hierarchical": false,
>   "type": "externalsource",
>   "_links": {
>     "self": {
>       "href": "https://myhost/server/api/integration/externalsources/openaireAPI"
>     },
>     "entries": {
>       "href": "https://myhost/server/api/integration/externalsources/openaireAPI/entries?query="
>     }
>   }
> }
> ```

I will investigate

**Comment by paulo-graca** [comment]
Hi @dimitrispie I'm having Spring errors when deploying the project:
```
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-02-12 16:45:33,619 ERROR org.springframework.boot.SpringApplication @ Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'externalSourcesRestController': Unsatisfied dependency expressed through field 'externalSourceRestRepository'; nested exception is org.sprin
gframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'integration.externalsource': Unsatisfied dependency expressed through field 'externalDataService'; nested exception is org.springframework.beans.fact
ory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.external.service.impl.ExternalDataServiceImpl#0': Unsatisfied dependency expressed through field 'externalDataProviders'; nested exception is org.springframewo
rk.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.external.provider.impl.OpenAIREFunderDataProvider#0' defined in file [/dspace/config/spring/api/external-services.xml]: Invocation of init method failed; n
ested exception is org.springframework.beans.factory.support.BeanDefinitionValidationException: Could not find an init method named 'init' on bean with name 'org.dspace.external.provider.impl.OpenAIREFunderDataProvider#0'
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:643) ~[spring-beans-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:130) ~[spring-beans-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:399) ~[spring-beans-5.2.5.RELEASE.jar:5.2.5.RELEASE]
        at 
...
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.dspace.external.service.impl.ExternalDataServiceImpl#0': Unsatisfied dependency expressed through field 'externalDataProviders'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.dspace.external.provider.impl.OpenAIREFunderDataProvider#0' defined in file [/dspace/config/spring/api/external-services.xml]: Invocation of init method failed; nested exception is org.springframework.beans.factory.support.BeanDefinitionValidationException: Could not find an init method named 'init' on bean with name 'org.dspace.external.provider.impl.OpenAIREFunderDataProvider#0'
```

**Comment by tdonohue** [comment]
Closing, replaced by #3365

**Pull request closed without merging**